src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,256 @@
+    return instant.atZone(ZoneId.ofOffset("UTC", timezoneOffset));
+  }
+
+  // This method takes the timezoneOffsetMinutes int and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlots(
+      TimeRange range, ZoneOffset timezoneOffset) {
+
+    Set<String> interviewers = availabilityDao.getUsersAvailableInRange(range.start(), range.end());
+    Set<PossibleInterviewSlot> possibleInterviews = new HashSet<PossibleInterviewSlot>();
+
+    // We don't want to schedule an interview for a user with themself, so we are removing
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
+    // replaced with this hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    interviewers.remove(userId);
+
+    for (String interviewer : interviewers) {
+      possibleInterviews.addAll(
+          getPossibleInterviewSlotsForPerson(interviewer, range, timezoneOffset));
+    }
+
+    // We need to check that the person looking to schedule is not already scheduled during any of
+    // the proposed times.
+
+    List<ScheduledInterview> userScheduledInterviews =
+        scheduledInterviewDao.getScheduledInterviewsInRangeForUser(
+            userId, range.start(), range.end());
+
+    Set<PossibleInterviewSlot> conflictingInterviews = new HashSet<PossibleInterviewSlot>();
+
+    for (PossibleInterviewSlot slot : possibleInterviews) {
+      for (ScheduledInterview userInterview : userScheduledInterviews) {
+        if (userInterview.when().contains(Instant.parse(slot.utcEncoding()))) {
+          conflictingInterviews.add(slot);
+        }
+      }
+    }
+
+    possibleInterviews.removeAll(conflictingInterviews);
+
+    List<PossibleInterviewSlot> possibleInterviewList =
+        new ArrayList<PossibleInterviewSlot>(possibleInterviews);
+
+    sortInterviews(possibleInterviewList);
+    return possibleInterviewList;
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlotsForPerson(
+      String userId, TimeRange range, ZoneOffset timezoneOffset) {
+    List<Availability> availabilities =
+        availabilityDao.getInRangeForUser(userId, range.start(), range.end());
+    availabilities.removeIf(avail -> avail.scheduled());
+
+    List<PossibleInterviewSlot> possibleInterviewSlotsForPerson =
+        new ArrayList<PossibleInterviewSlot>();
+    for (int i = 0; i < availabilities.size() - 3; i++) {
```

Comment:
This `3` needs to be made a constant.

No magic numbers please.

Reply:
Done